logexp/cmd/main.go

```diff
@@ -0,0 +1,257 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This script when run on a node (/master) in a kuberentes cluster would
+// export the logs to gcs, with the upload path and other config taken in
+// through environment variables.
+
+// TODO(shyamjvs): Make this work for master, currently facing gcloud auth
+// error when run from within a pod on the master.
+
+package main
+
+import (
+	goflag "flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/pflag"
+)
+
+// Initialize the log exporter's configuration related flags.
+var (
+	nodeName             string
+	gcsArtifactsPath     string
+	kubernetesProvider   string
+	gcloudAuthFilePath   string
+	enableHollowNodeLogs bool
+	sleepDuration        time.Duration
+)
+
+var (
+	// Source and destination logs directory.
+	localLogPath = "/var/log"
+	gcsLogPath = ""  // Computed as "gcsArtifactsPath/nodeName"
```

Comment:
No good reason, removed it now. Just wanted to make all config-related variables visible at the top.